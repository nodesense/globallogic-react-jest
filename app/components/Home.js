// state
import React from "react";
import {Text, View, Button} from 'react-native';

export default class Home extends React.Component {
    constructor(props) {
        super(props);

        // state is keyword for react
        // mutable
        this.state = {
            counter: 1000
        }
    }

    increment() {
       this.setState({
           counter: this.state.counter + 1
       })

       console.log("***INCR***")
    }

    decrement = () => {
        //console.trace();
        //keyword setState
        //GOOD
        // async
        // calls render
        console.log("before ", this.state.counter);
        this.setState( {
            counter : this.state.counter - 1
        })
 
        console.log("after ", this.state.counter);
    }

    reset = () => {
        console.log("***RESET");
        this.setState({
            counter: 0
        })
    }

    render() {
        console.log("Home render", this.state.counter);

        return (
            <View>
                <Text>Home</Text>
                <Text> Counter: {this.state.counter} </Text>

                <Button title="+1"
                        ref= { (o) => this.btn = o }
                        onPress={ () => this.increment() }>
                   
                </Button>

                {/* <Button title="-1"
                        onPress={this.decrement }>
                
                </Button>

                  <Button title="Reset"
                        onPress={this.reset }>
                
                </Button> */}
            </View>
        )
    }
}