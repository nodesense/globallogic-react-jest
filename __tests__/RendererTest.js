
import 'react-native';
import React from 'react';
import Home from '../app/components/Home';

import {shallow, mount} from "enzyme";
import { shallowToJson } from 'enzyme-to-json';


import {Text, View, Button} from 'react-native';


// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

fdescribe("Home test ", () => {
    
  xit('renders correctly', () => {
    const tree = renderer.create(
      <Home />
    );

    const treeJson = renderer.create(
      <Home />
    ).toJSON();


    console.log("JSON ", treeJson);
    expect(treeJson).toMatchSnapshot();

  });


  it('Home with increment', () => {
    const tree = renderer.create(
      <Home />
    );

    const treeJson = renderer.create(
      <Home />
    ).toJSON();

    let root = tree.root;

    //let instance = tree.getInstance()
    let instance = root.instance;

    instance.increment();
    instance.increment();
    instance.increment();
    // tree.update();

    //expect(treeJson).toMatchSnapshot();
    expect(instance.state.counter).toBe(1003);

    root.findByType(Button).props.onPress();

    expect(instance.state.counter).toBe(1004);

    let dump = tree.toJSON();
    expect(dump).toMatchSnapshot();
    root.findByType(Button).props.onPress();

    dump = tree.toJSON();
    expect(dump).toMatchSnapshot();

  });

});