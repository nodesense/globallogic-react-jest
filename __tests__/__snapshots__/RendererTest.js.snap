// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test  Home with increment 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Home
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
     Counter: 
    1004
     
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        +1
      </Text>
    </View>
  </View>
</View>
`;

exports[`Home test  Home with increment 2`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Home
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
     Counter: 
    1005
     
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        +1
      </Text>
    </View>
  </View>
</View>
`;
